### PR TITLE
refactor: refactor types

### PR DESCRIPTION
**Pull request by D-Sketon**
## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -121,9 +233,9 @@ export type NormalPageGenerator = {
...
 
 export interface SiteLocals {
-  posts: any; // _Query
-  pages: any; // _Query
-  categories: any; // _Model
-  tags: any; // _Model
+  posts: Query<PostSchema>; // _Query
+  pages: Query<PageSchema>; // _Query
+  categories: Model<CategorySchema>; // _Model
+  tags: Model<TagSchema>; // _Model
```
Comment (D-Sketon) [inline lib/types.ts:239]: after #5388, Model should be Query again

**Review by yoshinorin — COMMENTED** [review]
~~Shouldn’t we update the warehouse package's version from `^5.0.0` to `^5.0.1`?~~

~~Because this PR depends on `warehouse@5.0.1`. Of course we are using caret for specify version, it's means no worry with installation. But, IMHO we should explicitly specify package versions to clarify dependencies.~~

-----------------------

P.S:

Sorry, I didn't notice that...

https://github.com/hexojs/hexo/pull/5398/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Review by SukkaW — DISMISSED** [review]

**Review by uiolee — APPROVED** [review]

**Comment by D-Sketon** [comment]
fix #5403

**Review by SukkaW — COMMENTED** [review]
`any` would work for now. But IMHO we could use interface merging + theme overloading in the future.

**Review by SukkaW — APPROVED** [review]

**Pull request merged**